### PR TITLE
[Codestyle] Use @link over @see for URL's and use correct return types

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Use @link over @see for URL's and use correct return types in the doc blocks

### Testing Instructions

Code review as there a no production code changed

### Documentation Changes Required

None

### Additional Informations

I have excluded the externeal components such as fof (which has a lot of errors too) and everything in the vendor folder.

### CODE REVIEW

File: administrator/components/com_fields/models/fields.php [hunk 1]
```diff
@@ -350,7 +350,7 @@ protected function _getList($query, $limitstart = 0, $limit = 0)
 	 * @param   array    $data      data
 	 * @param   boolean  $loadData  load current data
 	 *
-	 * @return  JForm/false  the JForm object or false
+	 * @return  JForm / false  the JForm object or false
```
Comment (mbabker) [inline administrator/components/com_fields/models/fields.php]: Should be a pipe, `JForm|false`

File: build/phpcs/Joomla/Sniffs/Commenting/SingleCommentSniff.php [hunk 2]
```diff
@@ -5,7 +5,7 @@ class Joomla_Sniffs_Commenting_SingleCommentSniff implements PHP_CodeSniffer_Sni
 	/**
 	 * Returns the token types that this sniff is interested in.
 	 *
-	 * @return array(int)
+	 * @return array
```
Comment (mbabker) [inline build/phpcs/Joomla/Sniffs/Commenting/SingleCommentSniff.php]: If this is returning an array of integers the correct annotation would be `@return integer[]`

File: build/phpcs/Joomla/Sniffs/ControlStructures/ControlSignatureSniff.php [hunk 3]
```diff
@@ -45,7 +45,7 @@ public function __construct()
 	/**
 	 * Returns the patterns that this test wishes to verify.
 	 *
-	 * @return array(string)
+	 * @return array
```
Comment (mbabker) [inline build/phpcs/Joomla/Sniffs/ControlStructures/ControlSignatureSniff.php]: Same here, `@return string[]`

File: libraries/joomla/authentication/authentication.php [hunk 4]
```diff
@@ -319,7 +319,7 @@ public function authenticate($credentials, $options = array())
 	 * @param   JAuthenticationResponse  $response  response including username of the user to authorise
 	 * @param   array                    $options   list of options
 	 *
-	 * @return  array[JAuthenticationResponse]  results of authorisation
+	 * @return  array                    Array with results of authorisation containing a JAuthenticationResponse
```
Comment (mbabker) [inline libraries/joomla/authentication/authentication.php]: `@return JAuthenticationResponse[] Array of authentication response objects` (or something like that)

**Comment by zero-24** [comment]
Thanks @mbabker i have fixed the inline comments.

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 853e77dd1d88d25ec677968ec8dcb6632ae8fcfa<br /><br />review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15302">issues.joomla.org/tracker/joomla-cms/15302</a>.</sub>

**Comment by zero-24** [comment]
Thanks :+1:

**Pull request merged**